### PR TITLE
generate an OpenCL C built-in functions dictionary

### PR DESCRIPTION
**Pull request by bashbaug**
Possible fix for #1214 

This PR generalizes the script used to generate OpenCL C feature dictionaries from #1212 so it can be used for other types of dictionaries also.

Then:

1. It adds a list of OpenCL C built-in functions and uses the script to generate a dictionary for them.
2. Replaces the built-in functions with hard-coded unicode zero-width spaces with the asciidoctor attributes instead.

I believe this fixes the issue.

A few additional notes:

* I've only added the built-in functions with the hard-coded zero-width spaces.  If desired, we could add additional built-in functions, also.
* I haven't included any formatting in the generated built-in functions dictionary because some functions use **bold** formatting and other functions use `monospace` formatting.  We ought to unify this at some point.

I've tried copy-pasting the function names from the HTML spec with both Chrome and FireFox and they both seem to work fine.  I'm able to copy-paste the function names from the PDF spec, also.

### CODE REVIEW

**Comment by bashbaug** [comment]
I found a few formatting issues I need to fix.  Converting to draft.

**Comment by bashbaug** [comment]
The formatting issue is fixed, so this is now ready to go.

File: scripts/gen_dictionary_from_file.py [hunk 1]
```diff
@@ -11,40 +11,41 @@
 if __name__ == "__main__":
     parser = argparse.ArgumentParser()
 
-    parser.add_argument('-features', action='store',
+    parser.add_argument('-file', action='store',
```
Comment (kpet) [inline scripts/gen_dictionary_from_file.py:14]:
```suggestion
    parser.add_argument('-i', action='store',
```
or `-inputfile` maybe.

**Review by kpet — APPROVED** [review]

File: scripts/gen_dictionary_from_file.py (hunk 1 above)
Comment (bashbaug) [reply scripts/gen_dictionary_from_file.py]: Done - I switched to `-i` everywhere in the script, and the Makefile.

**Review by kpet — APPROVED** [review]

**Pull request merged**